eclair-core/src/test/scala/fr/acinq/eclair/router/ChannelRouterIntegrationSpec.scala

```diff
@@ -0,0 +1,124 @@
+package fr.acinq.eclair.router
+
+import akka.actor.ActorSystem
+import akka.testkit.{TestFSMRef, TestProbe}
+import fr.acinq.eclair.blockchain.bitcoind.ZmqWatcher
+import fr.acinq.eclair.blockchain.bitcoind.ZmqWatcher.WatchFundingDeeplyBuriedTriggered
+import fr.acinq.eclair.channel.DATA_NORMAL
+import fr.acinq.eclair.channel.states.{ChannelStateTestsBase, ChannelStateTestsTags}
+import fr.acinq.eclair.io.Peer.PeerRoutingMessage
+import fr.acinq.eclair.router.Router.{GossipOrigin, LocalGossip}
+import fr.acinq.eclair.wire.protocol.AnnouncementSignatures
+import fr.acinq.eclair.{BlockHeight, TestKitBaseClass}
+import org.scalatest.funsuite.FixtureAnyFunSuiteLike
+import org.scalatest.{Outcome, Tag}
+
+/**
+ * This test checks the integration between Channel and Router (events, etc.)
+ */
+class ChannelRouterIntegrationSpec extends TestKitBaseClass with FixtureAnyFunSuiteLike with ChannelStateTestsBase {
+
+  case class FixtureParam(router: TestFSMRef[Router.State, Router.Data, Router], rebroadcastListener: TestProbe, channels: SetupFixture, testTags: Set[String])
+
+  implicit val log: akka.event.LoggingAdapter = akka.event.NoLogging
+
+  override def withFixture(test: OneArgTest): Outcome = {
+    val channels = init(tags = test.tags)
+    val rebroadcastListener = TestProbe()
+    val router: TestFSMRef[Router.State, Router.Data, Router] = {
+      // we use alice's actor system so we share the same event stream
+      implicit val system: ActorSystem = channels.alice.underlying.system
+      system.eventStream.subscribe(rebroadcastListener.ref, classOf[Router.Rebroadcast])
+      TestFSMRef(new Router(channels.alice.underlyingActor.nodeParams, channels.alice.underlyingActor.blockchain, initialized = None))
+    }
+    withFixture(test.toNoArgTest(FixtureParam(router, rebroadcastListener, channels, test.tags)))
+  }
+
+  test("private local channel") { f =>
+    import f._
+
+    reachNormal(channels, testTags)
+
+    awaitCond(router.stateData.privateChannels.size == 1)
+
+    {
+      // only the local channel_update is known
+      val pc = router.stateData.privateChannels.values.head
+      assert(pc.update_1_opt.isDefined ^ pc.update_2_opt.isDefined)
```

Comment:
Why isn't Bob sending us his `channel_update`? Is it because it's not deeply buried yet? Can you add a comment to explain that?

Reply:
Indeed we currently only send the `channel_update` to our direct peer if the channel is deeply buried:
https://github.com/ACINQ/eclair/blob/30412766fdd701b03b4d5fc8922f42e9a7f131d6/eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala#L1596-L1610

But that looks like a bug to me... which has been fixed in #2224. That's why we now have this in the test setup: https://github.com/ACINQ/eclair/blob/8aee27a724cfef6f2b269e7ab37544b5dff77651/eclair-core/src/test/scala/fr/acinq/eclair/channel/states/ChannelStateTestsHelperMethods.scala#L244-L246

Reply:
I added a lame comment in a7a35c3bb, will get back to it in #2224 after the behavior changes.